sdk/identity/azure-identity/azure/identity/_internal/pipeline.py

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from azure.core.configuration import Configuration
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.policies import (
+    ContentDecodePolicy,
+    CustomHookPolicy,
+    DistributedTracingPolicy,
+    HeadersPolicy,
+    NetworkTraceLoggingPolicy,
+    ProxyPolicy,
+    RetryPolicy,
+    UserAgentPolicy,
+    HttpLoggingPolicy,
+)
+from azure.core.pipeline.transport import RequestsTransport
+
+from .user_agent import USER_AGENT
+
+
+def _get_config(**kwargs):
+    """Configuration common to a/sync pipelines"""
+    config = Configuration(**kwargs)
+    config.custom_hook_policy = CustomHookPolicy(**kwargs)
+    config.headers_policy = HeadersPolicy(**kwargs)
```

Comment:
Yes, this adds new policies to some pipelines. HeadersPolicy is a no-op absent configuration which I'm not adding here. Adding the policy could nonetheless break an application trying to set a header that causes the STS to reject a token request. Such an application might work today because we ignore its fail headers but eh, I can live with that risk 😸. ContentDecodePolicy is also new in some places, but I adjusted methods which deserialize responses to accommodate it. This is definitely something worth thinking through though, thanks for doing it, and let me know if anything else makes you go hmm. I may have missed something we don't have a test for.